arbnode/sequencer.go

```diff
@@ -284,15 +278,43 @@ func NewSequencer(txStreamer *TransactionStreamer, l1Reader *headerreader.Header
 		config:          configFetcher,
 		senderWhitelist: senderWhitelist,
 		nonceCache:      newNonceCache(config.NonceCacheSize),
-		nonceFailures:   containers.NewLruCacheWithOnEvict(config.NonceCacheSize, onNonceFailureEvict),
 		l1BlockNumber:   0,
 		l1Timestamp:     0,
 		pauseChan:       nil,
 	}
+	s.nonceFailures = containers.NewLruCacheWithOnEvict(config.NonceCacheSize, s.onNonceFailureEvict)
 	txStreamer.EnableReorgSequencing()
 	return s, nil
 }
 
+func (s *Sequencer) onNonceFailureEvict(_ addressAndNonce, failure *nonceFailure) {
+	if failure.revived {
+		return
+	}
+	queueItem := failure.queueItem
+	err := queueItem.ctx.Err()
+	if err != nil {
+		queueItem.returnResult(err)
+		return
+	}
+	_, forwarder := s.GetPauseAndForwarder()
```

Comment:
What do you think about moving GetPauseAndForwarder into the thread launched?
with that, you could call nonceFailures.Clear from the ForwardTo() fun, so messages will be forwarded immediately.

Reply:
I've implemented this in expireNonceFailures so we won't need a mutex